### PR TITLE
Add pre-built bindings, remove bindgen dep

### PR DESCRIPTION
**Pull request by vijk777**
Remove `bindgen` and `skeptic` dependencies and use the autogenerated bindings.rs unless feature=bindgen, following the rust-htslib model.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -1,50 +1,56 @@
...
-    bindings
-        .write_to_file(out_path.join("bindings.rs"))
-        .expect("Couldn't write bindings!");
-
-
-
-    // generates doc tests for `README.md`.
-    // skeptic::generate_doc_tests(&["README.md"]);
+    // If bindgen is enabled, use it
+    #[cfg(feature = "bindgen")]
```
Comment (pmarks) [inline build.rs]: you can only actually switch this on if you put a `bindgen` feature in the Cargo.toml.  My recommendation would be to remove this block & put a description of how to manually generate the bindings:

```cargo install bindgen
bindgen --no-doc-comments --whitelist-function mem_align1_core <more args mirroring this section> wrapper.h -o <appropriate bindings file>
```

We should probably also include a Mac version for completeness (or test that the generated files work on Mac)

File: Cargo.toml [hunk 2]
```diff
@@ -9,11 +9,4 @@ build = "build.rs"
...
 failure_derive = "*"
-rust-htslib = { git = "https://github.com/rust-bio/rust-htslib.git", rev = "f149f9f6172d26b5921693e84f4334d7108f1814", default-features = false, features = ["serde_feature"] }
-
-[build-dependencies]
-bindgen = ">=0.47"
-skeptic = "*"
-
-[dev-dependencies]
-skeptic = "*"
+rust-htslib = { git = "https://github.com/rust-bio/rust-htslib.git", rev = "f149f9f6172d26b5921693e84f4334d7108f1814", default-features = false, features = ["serde_feature"] }
```
Comment (pmarks) [inline Cargo.toml]: everything we need is now in the official release, so let's take this opportunity to update:
```suggestion
rust-htslib = { version = "0.34.0", default-features = false, features = ["serde_feature"] }
```

File: build.rs [hunk 3]
```diff
@@ -1,50 +1,56 @@
+#[cfg(feature = "bindgen")]
```
Comment (pmarks) [inline build.rs]: remove

**Review by pmarks — COMMENTED** [review]
Nice! A few small nits & we're good to go.  I think we should just switch wholesale to manually generated bindings and remove all mentions of bindgen, except for something in the readme showing how to generate them on the cmd-line (see below).

File: build.rs (hunk 1 above)
Comment (vijk777) [reply build.rs]: I did not realize you could do this from the command line. Yes, will get rid of this.

**Pull request merged**